### PR TITLE
Update documentation for Kubeapps 1.8

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Update several files to reflect the latest status of Kubeapps (and Helm). This PR is better merged after the new release (1.8.0) is available for users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,12 +15,16 @@ Kubeapps is a web-based UI for deploying and managing applications in Kubernetes
...
 
+***Note:*** Kubeapps supports both Helm v2 and v3. For detailed instructions on how to install and use Kubeapps follow the [Getting Started Guide](docs/user/getting-started.md).
+
 ```bash
 helm repo add bitnami https://charts.bitnami.com/bitnami
+# Helm v2 command
+helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
+# Helm v3 command
+kubectl create namespace kubeapps
 helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
```
Comment (absoludity) [inline README.md]: Looks like a copy-n-pasto, as `--name` isn't supported on the helm3 cli:
```
$ helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
Error: unknown flag: --name
```

Also, we need the `--useHelm3` flag. (EDIT: exactly like you have in the chart readme below)

File: chart/kubeapps/README.md [hunk 2]
```diff
@@ -34,19 +44,26 @@ It also packages the [Bitnami MongoDB chart](https://github.com/helm/charts/tree
...
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm install kubeapps --namespace kubeapps bitnami/kubeapps --set useHelm3=true
+```
 
 The command deploys Kubeapps on the Kubernetes cluster in the `kubeapps` namespace. The [Parameters](#parameters) section lists the parameters that can be configured during installation.
 
 > **Caveat**: Only one Kubeapps installation is supported per namespace
 
-> **Tip**: List all releases using `helm list`
-
```
Comment (absoludity) [inline chart/kubeapps/README.md]: Why did you remove this? If we add the `--all-namespaces` flag, it still lists releases across namespaces (or the `--all` flag for all statuses).

File: docs/developer/kubeops.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Kubeapps Kubeops Developer Guide
+
+The `kubeops` component is a micro-service that creates a API endpoint for accessing the Helm API and Kubernetes resources.
```
Comment (absoludity) [inline docs/developer/kubeops.md]: `an API endpoint`

File: chart/kubeapps/README.md (hunk 2 above)
Comment (andresmgot) [reply chart/kubeapps/README.md]: I removed it because the command is different for helm 2 and helm 3 and I didn't want to add more clutter to the comment (also the tip is not helping much)

File: chart/kubeapps/values.yaml [hunk 4]
```diff
@@ -8,6 +8,10 @@
 #     - myRegistryKeySecretName
 #   storageClass: myStorageClass
 
+## Enable the useHelm3 feature flag to use Kubeapps with Helm3 support.
```
Comment (SimonAlling) [inline chart/kubeapps/values.yaml]:
```suggestion
## Enable this feature flag to use Kubeapps with Helm 3 support.
```
Suggesting the first change because it's more DRY and the second one because it's (more) correct, I think. At least the Helm people [use that style themselves](https://helm.sh/blog/helm-3-released/).

File: chart/kubeapps/values.yaml [hunk 5]
```diff
@@ -8,6 +8,10 @@
...
+## If you set it to true, Kubeapps will not work with Helm2 releases.
```
Comment (SimonAlling) [inline chart/kubeapps/values.yaml]:
```suggestion
## If you set it to true, Kubeapps will not work with releases installed with Helm 2.
```
I think this might be more clear, so people don't think they can't use Helm 2 _charts_ with `useHelm3: true`.

File: docs/architecture/overview.md [hunk 6]
```diff
@@ -12,12 +12,17 @@ Additionally, the dashboard integrates with the [Kubernetes service catalog](htt
...
-### Tiller proxy
+### Tiller proxy (deprecated)
 
 In order to secure the access to Tiller and allow the dashboard to contact the Helm Tiller server we deploy a proxy that handles the communication with Tiller. The goal of this proxy is to validate that the user doing the request has sufficent permissions to create or delete all the resources that are part of the specific chart being deployed or deleted.
 
 This proxy is written in Go. Check more details about the implementation in this [document](/cmd/tiller-proxy/README.md).
 
+### Kubeops
+
+Kubeops is the successor of Tiller proxy. It's the service in charge of comunicating both with the Helm (v3) API and other k8s resources like AppRepositories or Secrets.
```
Comment (SimonAlling) [inline docs/architecture/overview.md]:
```suggestion
Kubeops is the successor of Tiller proxy. It's the service in charge of communicating both with the Helm (v3) API and other k8s resources like AppRepositories or Secrets.
```

File: docs/architecture/overview.md [hunk 7]
```diff
@@ -12,12 +12,17 @@ Additionally, the dashboard integrates with the [Kubernetes service catalog](htt
...
+Check more details about the implementation in this [document](/docs/developer/kubeops.md).
```
Comment (SimonAlling) [inline docs/architecture/overview.md]:
```suggestion
Check more details about the implementation in [this document](/docs/developer/kubeops.md).
```

File: docs/developer/kubeops.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Environment
+
+```bash
+export GOPATH=~/gopath
+export PATH=$GOPATH/bin:$PATH
+export KUBEAPPS_DIR=$GOPATH/src/github.com/kubeapps/kubeapps
+```
+## Download the kubeapps source code
```
Comment (SimonAlling) [inline docs/developer/kubeops.md]:
```suggestion
## Download the Kubeapps source code
```

File: docs/developer/kubeops.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+git clone --recurse-submodules https://github.com/kubeapps/kubeapps $KUBEAPPS_DIR
+```
+
+The `kubeops` sources are located under the `cmd/kubeops/` and it uses packages from the `pkg` directory.
```
Comment (SimonAlling) [inline docs/developer/kubeops.md]:
```suggestion
The `kubeops` sources are located under `cmd/kubeops/` and use packages from the `pkg` directory.
```

File: docs/developer/kubeops.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+eval $(minikube docker-env)
+```
+
+Note: By default, Kubeapps will try to fetch the latest version of the image so in order to make this workflow work in Minikube you will need to update the imagePullPolicy first:
+
+```
+kubectl patch deployment kubeapps-internal-kubeops -n kubeapps --type=json -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/imagePullPolicy", "value": "IfNotPresent"}]'
+```
+
+The easiest way to create the `kubeops` image is execute the Makefile task to do so:
```
Comment (SimonAlling) [inline docs/developer/kubeops.md]:
```suggestion
The easiest way to create the `kubeops` image is to execute the Makefile task to do so:
```

File: docs/developer/kubeops.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+kubectl set image -n kubeapps deployment kubeapps-internal-kubeops kubeops=kubeapps/kubeops:dev
+```
+
+For further redeploys you can change the version to deploy a different tag or rebuild the same image and restart the pod executing:
+
+```
+kubectl delete pod -n kubeapps -l app=kubeapps-internal-kubeops
+```
+
+Note: If you using a cloud provider to develop the service you will need to retag the image and push it to a public registry.
```
Comment (SimonAlling) [inline docs/developer/kubeops.md]:
```suggestion
Note: If you are using a cloud provider to develop the service you will need to retag the image and push it to a public registry.
```

File: docs/developer/kubeops.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+### Running tests
+
+You can run the kubeops tests along with the tests of all the projecs
```
Comment (SimonAlling) [inline docs/developer/kubeops.md]:
```suggestion
You can run the kubeops tests along with the tests of all the projects:
```

File: docs/user/getting-started.md [hunk 13]
```diff
@@ -17,6 +17,13 @@ helm repo add bitnami https://charts.bitnami.com/bitnami
 helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
 ```
 
+If you are using Helm v3, you need to set an extra flag to enable it:
```
Comment (SimonAlling) [inline docs/user/getting-started.md]:
```suggestion
If you are using Helm 3, you need to set an extra flag to enable it:
```

File: docs/user/securing-kubeapps.md [hunk 14]
```diff
@@ -1,17 +1,17 @@
-# Securing Kubeapps installation
+# Securing Kubeapps installation for Helm v2
```
Comment (SimonAlling) [inline docs/user/securing-kubeapps.md]:
```suggestion
# Securing Kubeapps installation for Helm 2
```

File: docs/user/securing-kubeapps.md [hunk 15]
```diff
@@ -1,17 +1,17 @@
...
 
-In this guide we will explain how to secure the installation of Kubeapps in a multi-tenant cluster. Following these steps are only necessary if different people with different permissions have access to the same cluster. Generic instructions to secure Helm can be found [here](https://github.com/kubernetes/helm/blob/master/docs/securing_installation.md).
+In this guide we will explain how to secure the installation of Kubeapps in a multi-tenant cluster if you are still using the version 2 of Helm. Following these steps are only necessary if different people with different permissions have access to the same cluster. Generic instructions to secure Helm can be found [here](https://github.com/helm/helm/blob/dev-v2/docs/securing_installation.md).
```
Comment (SimonAlling) [inline docs/user/securing-kubeapps.md]:
```suggestion
In this guide we will explain how to secure the installation of Kubeapps in a multi-tenant cluster if you are still using version 2 of Helm. Following these steps are only necessary if different people with different permissions have access to the same cluster. Generic instructions to secure Helm can be found [here](https://github.com/helm/helm/blob/dev-v2/docs/securing_installation.md).
```

File: chart/kubeapps/README.md [hunk 16]
```diff
@@ -13,16 +13,26 @@
 
 ## TL;DR;
 
+For Helm v2:
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
For Helm 2:
```

File: chart/kubeapps/README.md [hunk 17]
```diff
@@ -13,16 +13,26 @@
...
+
 ```bash
 helm repo add bitnami https://charts.bitnami.com/bitnami
 helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
 ```
 
+For Helm v3:
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
For Helm 3:
```

File: chart/kubeapps/README.md [hunk 18]
```diff
@@ -34,19 +44,26 @@ It also packages the [Bitnami MongoDB chart](https://github.com/helm/charts/tree
 
 To install the chart with the release name `kubeapps`:
 
+For Helm v2:
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
For Helm 2:
```

File: chart/kubeapps/README.md [hunk 19]
```diff
@@ -34,19 +44,26 @@ It also packages the [Bitnami MongoDB chart](https://github.com/helm/charts/tree
...
+
 ```console
 $ helm repo add bitnami https://charts.bitnami.com/bitnami
 $ helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
 ```
 
-> **IMPORTANT** This assumes an insecure Helm installation, which is not recommended in production. See [the documentation to learn how to secure Helm and Kubeapps in production](https://github.com/kubeapps/kubeapps/blob/master/docs/user/securing-kubeapps.md).
+> **IMPORTANT** This assumes an insecure Helm v2 installation, which is not recommended in production. See [the documentation to learn how to secure Helm v2 and Kubeapps in production](https://github.com/kubeapps/kubeapps/blob/master/docs/user/securing-kubeapps.md).
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
> **IMPORTANT** This assumes an insecure Helm 2 installation, which is not recommended in production. See [the documentation to learn how to secure Helm 2 and Kubeapps in production](https://github.com/kubeapps/kubeapps/blob/master/docs/user/securing-kubeapps.md).
```

File: chart/kubeapps/README.md [hunk 20]
```diff
@@ -34,19 +44,26 @@ It also packages the [Bitnami MongoDB chart](https://github.com/helm/charts/tree
...
+
+For Helm v3:
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
For Helm 3:
```

File: chart/kubeapps/README.md [hunk 21]
```diff
@@ -81,13 +98,13 @@ Kubeapps supports two database types: MongoDB or PostgreSQL. By default MongoDB
 
 > **Note**: Changing the database type when upgrading is not supported.
 
-### Configuring connection to a custom namespace Tiller instance
+### [Only for Helm v2] Configuring connection to a custom namespace Tiller instance
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
### [Only for Helm 2] Configuring connection to a custom namespace Tiller instance
```

File: chart/kubeapps/README.md [hunk 22]
```diff
@@ -81,13 +98,13 @@ Kubeapps supports two database types: MongoDB or PostgreSQL. By default MongoDB
...
 
 By default, Kubeapps connects to the Tiller Service in the `kube-system` namespace, the default install location for Helm.
 
 If your instance of Tiller is running in a different namespace or you want to have different instances of Kubeapps connected to different Tiller instances, you can achieve it by setting the `tillerProxy.host` parameter. For example, you can set `tillerProxy.host=tiller-deploy.my-custom-namespace:44134`
 
-### Configuring connection to a secure Tiller instance
+### [Only for Helm v2] Configuring connection to a secure Tiller instance
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
### [Only for Helm 2] Configuring connection to a secure Tiller instance
```

File: chart/kubeapps/README.md [hunk 23]
```diff
@@ -155,8 +170,11 @@ If you find issues upgrading Kubeapps, check the [troubleshooting](#error-while-
 To uninstall/delete the `kubeapps` deployment:
 
 ```console
+# For Helm v2
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
# For Helm 2
```

File: chart/kubeapps/README.md [hunk 24]
```diff
@@ -155,8 +170,11 @@ If you find issues upgrading Kubeapps, check the [troubleshooting](#error-while-
...
 $ helm delete --purge kubeapps
-$ # Optional: Only if there are no more instances of Kubeapps
+# For Helm v3
```
Comment (SimonAlling) [inline chart/kubeapps/README.md]:
```suggestion
# For Helm 3
```

File: README.md [hunk 25]
```diff
@@ -15,12 +15,16 @@ Kubeapps is a web-based UI for deploying and managing applications in Kubernetes
 
 Use the Helm chart to install the latest version of Kubeapps: 
 
+***Note:*** Kubeapps supports both Helm v2 and v3. For detailed instructions on how to install and use Kubeapps follow the [Getting Started Guide](docs/user/getting-started.md).
```
Comment (SimonAlling) [inline README.md]:
```suggestion
***Note:*** Kubeapps supports both Helm 2 and Helm 3. For detailed instructions on how to install and use Kubeapps follow the [Getting Started Guide](docs/user/getting-started.md).
```
This [seems to be the official style](https://helm.sh/blog/helm-3-released/). Hence/also, I think it would improve `Ctrl+F`-ability?  WDYT?

File: README.md [hunk 26]
```diff
@@ -15,12 +15,16 @@ Kubeapps is a web-based UI for deploying and managing applications in Kubernetes
...
+
 ```bash
 helm repo add bitnami https://charts.bitnami.com/bitnami
+# Helm v2 command
```
Comment (SimonAlling) [inline README.md]:
```suggestion
# Helm 2 command
```

File: README.md [hunk 27]
```diff
@@ -15,12 +15,16 @@ Kubeapps is a web-based UI for deploying and managing applications in Kubernetes
...
 helm install --name kubeapps --namespace kubeapps bitnami/kubeapps
+# Helm v3 command
```
Comment (SimonAlling) [inline README.md]:
```suggestion
# Helm 3 command
```
